buf.gen.ts.yaml

```diff
@@ -1,11 +1,24 @@
-version: v1
+# Learn more: https://buf.build/docs/configuration/v2/buf-gen-yaml
+version: v2
+clean: true
 plugins:
-  - name: ts
+  - local: ./ts/node_modules/.bin/protoc-gen-es
+    out: ts/src/gen/protos
     strategy: all
-    path: ./ts/node_modules/.bin/protoc-gen-ts_proto
-    out: ./ts/src/generated
-    opt: "esModuleInterop=true,forceLong=long,outputTypeRegistry=true,useExactTypes=false,exportCommonSymbols=false"
-  - name: grpc-gateway-ts
-    path: ./ts/node_modules/.bin/protoc-gen-ts_proto
-    out: ./.cache/tmp/ts/generated-grpc-js
-    opt: "esModuleInterop=true,forceLong=long,useExactTypes=false,exportCommonSymbols=false,outputServices=grpc-js"
\ No newline at end of file
+    include_imports: true
+    include_wkt: true
+    opt:
+      - target=ts
+      - json_types=true
+  - local: ./ts/bin/protoc-sdk-object.ts
+    out: ts/src/gen
+    strategy: all
+    include_imports: true
+    include_wkt: true
+    opt:
+      - target=ts
+
+inputs:
+  - module: buf.build/cosmos/cosmos-sdk
```

Comment:
the version of cosmos SDK has to be pinned, otherwise ts and go version will generate incompatible code. [buf.yaml](https://github.com/akash-network/akash-api/blob/sdk-47/proto/node/buf.yaml) within node already does it and this line need to be removed `- module: buf.build/cosmos/cosmos-sdk`

Reply:
this is odd as buf does it for go, so we need to figure out why ts part does not, .cache types have to be present due to k8s proto dependencies

Reply:
No it doesn't, in case of go very likely it just uses prebuilt files from cosmos-sdk library installed for go. You can run 

```
 .cache/bin/buf ls-files --include-import
```

to see what files are part of a workspace and will be processed by buf gen plugins.